### ISSUE TITLE
Apply UnitGenerator for record type

### ISSUE THREAD

**Issue by jetloong** [Cysharp/UnitGenerator#9]
First of all thanks for having this project. I found it useful.
I am mostly using the implicit-operator for domain specific value objects which are represented by one property e.g. DeviceIdentifier and OrderIdentifier where both are described just by a property of type string. But it much easier to reason and read when the abstraction is used in code. 

I like the idea of the new records-keyword. Is there a plan to support record-type?

**Comment by neuecc** [comment]
What do you think the benefit?

record automatically implements IEquatable and ToString.
UnitGenerator already has that functionality.
It also does not need to support multiple properties like a record does.

**Comment by jetloong** [comment]
In some projects I have started to use records and quite found it useful. 
In cases when there is just one property, I have added implicit-operators to simplify assigning or reading the one property. 
So my benefit would just be taking advantage of a sourcegenerator and writing less code. 
May be it is bad idea of me to work with records for value objects and I can just use struct as supported by UnitGenerator.

> It also does not need to support multiple properties like a record does.
I did not get your idea of that.
